virtualbox/resource_vm.go

```diff
@@ -525,8 +528,8 @@ func waitUntilVMIsReady(ctx context.Context, d *schema.ResourceData, vm *vbox.Ma
 			[]string{"no"},
 			key,
 			meta,
-			30*time.Second,
-			1*time.Second,
+			time.Duration(config.Delay)*time.Second, // used to be hardcoded as 30
```

Comment:
If the Delay itself is a time.Duration we will not need to convert it to time.Duration*time.Second here.